bundles/io/org.eclipse.smarthome.io.voice/src/main/java/org/eclipse/smarthome/io/voice/text/AbstractRuleBasedInterpreter.java

```diff
@@ -476,40 +568,19 @@ protected String executeSingle(ResourceBundle language, String[] nameFragments,
     /**
      * Tokenizes text. Filters out all unsupported punctuation. Tokens will be lower case.
      *
+     * @param locale the locale that should be used for lower casing
      * @param text the text that should be tokenized
      * @return resulting tokens
      */
-    protected ArrayList<String> tokenize(String text) {
-        ArrayList<String> parts = new ArrayList<String>();
-        String[] split = text.toLowerCase().replaceAll("[^\\w\\s]", " ").split("\\s");
-        for (int i = 0; i < split.length; i++) {
-            String part = split[i].trim();
-            if (part.length() > 0) {
-                parts.add(part);
-            }
-        }
-        return parts;
-    }
-
-    /**
-     * Splits an item's name into single words. It splits whitespace, Pascal, Camel and Snake-casing.
-     *
-     * @param name the name that's to be split
-     * @param toLowerCase if {true}, all resulting fragments will be made lower case
-     * @return resulting fragments of the name
-     */
-    protected ArrayList<String> splitName(String name, boolean toLowerCase) {
+    protected ArrayList<String> tokenize(Locale locale, String text) {
         ArrayList<String> parts = new ArrayList<String>();
-        if (name == null) {
+        if (text == null) {
             return parts;
         }
-        String[] split = name.replaceAll("\\.", "").split("(?<!^)(?=[A-Z])|_|\\s+");
+        String[] split = text.toLowerCase(locale).replaceAll("[\\']", "").replaceAll("[^\\w\\s]", " ").split("\\s");
```

Comment:
Just checking: is it intentional that you changed the first replaceAll from \\. to \\' ?

It might be worth more comments explaining why ' is filtered out and other punctuation characters like " are not.

Also, I suspect if you change the `.split("\\s")` to `.trim().split("\\s+")` then you won't need the for loop below.  And if you do that, the your method could return String[] instead of List<String>, though I'm not sure if you'd want that